app/src/main/java/com/android/unio/ui/event/overlay/PictureOverlay.kt

```diff
@@ -0,0 +1,113 @@
+package com.android.unio.ui.event.overlay
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.PagerState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.FilterQuality
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import androidx.core.net.toUri
+import com.android.unio.R
+import com.android.unio.model.event.EventUserPicture
+import com.android.unio.model.strings.test_tags.event.EventDetailsTestTags
+import com.android.unio.model.strings.test_tags.event.EventDetailsTestTags.PICTURE_FULL_SCREEN
+import com.android.unio.ui.image.AsyncImageWrapper
+import kotlinx.coroutines.launch
+
+/**
+ * A dialog that allows users to view event pictures in full screen.
+ *
+ * @param onDismiss Callback when the dialog is dismissed.
+ * @param pagerState the [PagerState] of the pager.
+ * @param eventPictures The list of [EventUserPicture] to display.
+ */
+@Composable
+fun PictureOverlay(
+    onDismiss: () -> Unit,
+    pagerState: PagerState,
+    eventPictures: List<EventUserPicture>
+) {
+  val scope = rememberCoroutineScope()
+  val context = LocalContext.current
+  val iconSize = 40.dp
+  val onClickArrow = { isRight: Boolean ->
+    if (isRight && pagerState.currentPage < eventPictures.size - 1) {
+      scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
+    } else if (!isRight) {
+      if (pagerState.currentPage > 0) {
+        scope.launch { pagerState.animateScrollToPage(pagerState.currentPage - 1) }
+      } else {}
+    } else {}
```

Comment:
Empty else statement. Consider handle these cases properly

Reply:
I did this because android studio was giving me the warning "'if' must have both main and 'else' branches if used as an expression" when I didn't put any else statement. But found that casting onClickArrow to (Boolean) -> Unit achieves the same and is cleaner than empty else statements :)